File: src/main/java/org/jcodec/common/Format.java
package org.jcodec.common;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * The list of formats known to JCodec
 * 
 * @author The JCodec project
 * 
 */
public final class Format {
    public final static Format MOV = new Format("MOV", true, true);
    public final static Format MPEG_PS = new Format("MPEG_PS", true, true);
    public final static Format MPEG_TS = new Format("MPEG_TS", true, true);
    public final static Format MKV = new Format("MKV", true, true);
    public final static Format H264 = new Format("H264", true, false);
    public final static Format RAW = new Format("RAW", true, true);
    public final static Format FLV = new Format("FLV", true, true);
    public final static Format AVI = new Format("AVI", true, true);
    public final static Format IMG = new Format("IMG", true, false);
    public final static Format IVF = new Format("IVF", true, false);
    public final static Format MJPEG = new Format("MJPEG", true, false);
    public final static Format Y4M = new Format("Y4M", true, false);
    public final static Format WAV = new Format("WAV", false, true);
    public final static Format WEBP = new Format("WEBP", true, false);
    public final static Format MPEG_AUDIO = new Format("MPEG_AUDIO", false, true);

    private final static Map<String, Format> _values = new LinkedHashMap<String, Format>();
    static {
        _values.put("MOV", MOV);
        _values.put("MPEG_PS", MPEG_PS);
        _values.put("MPEG_TS", MPEG_TS);
        _values.put("MKV", MKV);
        _values.put("H264", H264);
        _values.put("RAW", RAW);
        _values.put("FLV", FLV);
        _values.put("AVI", AVI);
        _values.put("IMG", IMG);
        _values.put("IVF", IVF);
        _values.put("MJPEG", MJPEG);
        _values.put("Y4M", Y4M);
        _values.put("WAV", WAV);
        _values.put("WEBP", WEBP);
        _values.put("MPEG_AUDIO", MPEG_AUDIO);
    }

    private final boolean video;
    private final boolean audio;
    private final String name;

    private Format(String name, boolean video, boolean audio) {
        this.name = name;
        this.video = video;
        this.audio = audio;
    }

    public boolean isAudio() {
        return audio;
    }

    public boolean isVideo() {
        return video;
    }

    public static Format valueOf(String s) {
        return _values.get(s);
    }
}

File: src/main/java/org/jcodec/common/Format.kt
package org.jcodec.common

import java.util.*

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * The list of formats known to JCodec
 *
 * @author The JCodec project
 */
class Format private constructor(private val name: String, val isVideo: Boolean, val isAudio: Boolean, val isContained: Boolean) {
    companion object {
        @JvmField
        val MOV = Format("MOV", true, true, true)
        @JvmField
        val MPEG_PS = Format("MPEG_PS", true, true, true)
        @JvmField
        val MPEG_TS = Format("MPEG_TS", true, true, true)
        @JvmField
        val MKV = Format("MKV", true, true, true)
        @JvmField
        val H264 = Format("H264", true, false, true)
        @JvmField
        val RAW = Format("RAW", true, true, true)
        @JvmField
        val FLV = Format("FLV", true, true, true)
        @JvmField
        val AVI = Format("AVI", true, true, true)
        @JvmField
        val IMG = Format("IMG", true, false, false)
        @JvmField
        val IVF = Format("IVF", true, false, true)
        @JvmField
        val MJPEG = Format("MJPEG", true, false, true)
        @JvmField
        val Y4M = Format("Y4M", true, false, true)
        @JvmField
        val WAV = Format("WAV", false, true, true)
        @JvmField
        val WEBP = Format("WEBP", true, false, true)
        @JvmField
        val MPEG_AUDIO = Format("MPEG_AUDIO", false, true, true)
        @JvmField
        val DASH = Format("DASH", true, true, false)
        @JvmField
        val DASHURL = Format("DASHURL", true, true, false)
        private val _values: MutableMap<String, Format> = LinkedHashMap()
        @JvmStatic
        fun valueOf(s: String): Format? {
            return _values[s]
        }

        init {
            _values["MOV"] = MOV
            _values["MPEG_PS"] = MPEG_PS
            _values["MPEG_TS"] = MPEG_TS
            _values["MKV"] = MKV
            _values["H264"] = H264
            _values["RAW"] = RAW
            _values["FLV"] = FLV
            _values["AVI"] = AVI
            _values["IMG"] = IMG
            _values["IVF"] = IVF
            _values["MJPEG"] = MJPEG
            _values["Y4M"] = Y4M
            _values["WAV"] = WAV
            _values["WEBP"] = WEBP
            _values["MPEG_AUDIO"] = MPEG_AUDIO
            _values["DASH"] = DASH
            _values["DASHURL"] = DASHURL
        }
    }

}